web-common/src/features/entity-management/file-artifact.ts

```diff
@@ -111,12 +111,12 @@ export class FileArtifact {
   }
 
   updateRemoteContent = (newContent: string, alert = true) => {
-    this.remoteContent.set(newContent);
     if (alert && newContent !== get(this.remoteContent)) {
       for (const callback of this.remoteCallbacks) {
         callback(newContent);
       }
     }
+    this.remoteContent.set(newContent);
```

Comment:
This makes me a little nervous – semantically, the callbacks should happen _after_ the remoteContent changes not _before_.

Reply:
Good point. `newContent !== get(this.remoteContent)` check should happen before updating content and firing callback should happen after.

Reply:
Nice, I really like the new approach